### ISSUE TITLE
"Microsoft.Common.props" not found when compiling Mono project, referencing netstandard2.0 project

### ISSUE THREAD

**Issue by JulianGmp** [godotengine/godot#25236]
**Godot version:** v3.0.6 win64

**Mono Version:** 5.12.0301 x64

**OS/device including version:** Windows 10 x64, Version 1803, Build 17134.523

**Issue description:** When implementing a C# classlibrary under netstandard2.0, compilation will fail, saying that `C:\Program Files (x86)\MSBuild\15.0\Microsoft.Common.props` was not found. (Full output below)

**Steps to reproduce:**
1. create new godot project, and attach a C# script to initialize the sln.
2. Create a classlib project using netstandard2.0, for example
`dotnet new classlib -o Lib1` (using .net core CLI)
3. Add new library project to solution, either through visual studio or
`dotnet sln add ./Lib1/Lib1.csproj`
4. Add a reference to the classlib project in your main csproj (that was generated from godot). It's easiest to use visual studio, or another IDE here, because of the stupidly long "id" the project gets, although you could copy that from the `sln` file.
```xml
<ProjectReference Include="Lib1\Lib1.csproj">
      <Project>{0fbfd7f2-b439-492f-99e4-842c6e3ddfa9}</Project>
      <Name>Lib1</Name>
</ProjectReference>
```
5. Try to compile through godot

After talking to @neikeq on Discord, I tried manually compiling in the mono command prompt with 
`msbuild aaa.sln /v:detailed /t:Build "/p:Configuration=Tools" "/l:GodotBuildLogger,D:\Users\julian\Desktop\Godot_v3.0.6-stable_mono_win64\GodotSharpTools.dll;logger/output/dir"`
This explicitly leaves out `/p:FrameworkPathOverride=mono_root/lib/mono/4.5`.

The entire output of the `msbuild` command I ran:
[msbuild_error_output.txt](https://github.com/godotengine/godot/files/2783816/msbuild_error_output.txt)

**Minimal reproduction project:** [aaa.zip](https://github.com/godotengine/godot/files/2783833/aaa.zip)

Please note that, according to [Microsoft's .NET Standard Documentation](https://docs.microsoft.com/en-us/dotnet/standard/net-standard), the minimum .NET Framework version for use with netstandard 2.0 is **v4.6.1**, and Godot initializes the project with **v4.5**. 
I tested using v4.6.1 and v4.7.2, but that did **not** change the errors I was getting from msbuild. 
I think this may be part of the problem.

**Comment by David-Ochoa** [comment]
This is a known error with Visual Studio Build Tools, you need to install full Visual Studio 2017, I tested it with 3.0.6 and 3.1 Beta 2 and compiles after changing target version to 4.6.1.

**Comment by JulianGmp** [comment]
Could you be a bit more specific with this?
I specifically installed VS Build Tools 2017 as you said, and included the .net build tools, but I end up with the same issue when compiling using the mono prompt and `msbuild`.
It does work fine if I use `dotnet build`. (Although I don't know how I would launch the game in the end, because I only have DLL files)

**Comment by David-Ochoa** [comment]
VS Build Tools 2017 lack some components, you need Visual Studio 2017 IDE (Community works) with all the required targets, using only VS Build Tools won't work.

**Comment by JulianGmp** [comment]
I had Visual Studio 2017 installed the entire time. And recently updated as well.

**Comment by David-Ochoa** [comment]
In my system I had MSBuild (VS Build Tools) and changing to the default MSBuild (Mono) raised the same problem, I'd suggest using this setting, but if you still want to use mono you'd need to tinker with "C:\Program Files\Mono\lib\mono\msbuild\15.0\bin\MSBuild.dll.config" file so mono can find the targets.

**Comment by JulianGmp** [comment]
So, I just let the explorer search for `Microsoft.Common.props`.
I found that I had several versions of it, all for v14 and below.
But one was downloaded through vscode, because I'm using the C# extension, which utilises Omnisharp.

So I just copied
`D:\Users\julian\.vscode\extensions\ms-vscode.csharp-1.17.1\.omnisharp\1.32.10-beta.5\msbuild\15.0\Microsoft.Common.Props` to `C:\Program Files (x86)\MSBuild\15.0\Microsoft.Common.Props`.

Compiling worked, and the project launched normally.
Not the fanciest workarounds, but it'll work.
